src/reducers/CoreReducer.ts

```diff
@@ -52,5 +54,32 @@ export const core = combineReducers<CoreState, Action>({
       default:
         return state
     }
+  },
+
+  enabledDetectedTokens(state: { [walletId: string]: string[] } = {}, action: Action): { [walletId: string]: string[] } {
+    switch (action.type) {
+      case 'CORE/NEW_TOKENS': {
+        const newState = { ...state }
+        const { walletId, enablingTokenIds } = action.data
+
+        if (enablingTokenIds.length === 0) showError(`enabledDetectedTokens: Received empty tokens array for ${walletId}`)
```

Comment:
I don't see why this test is needed. Was it for debugging purposes?

Reply:
No, it's an assertion